impeller/renderer/backend/vulkan/pipeline_cache_data_vk.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_IMPELLER_RENDERER_BACKEND_VULKAN_PIPELINE_CACHE_DATA_VK_H_
+#define FLUTTER_IMPELLER_RENDERER_BACKEND_VULKAN_PIPELINE_CACHE_DATA_VK_H_
+
+#include "flutter/fml/mapping.h"
+#include "flutter/fml/unique_fd.h"
+#include "impeller/renderer/backend/vulkan/vk.h"
+
+namespace impeller {
+
+//------------------------------------------------------------------------------
+/// @brief      Persist the pipeline cache to a file in the given cache
+///             directory. This function performs integrity checks the Vulkan
+///             driver may have missed.
+///
+/// @warning    The pipeline cache must be externally synchronized for most
```

Comment:
I left this in warning in there because it is theoretically possible if we are still in the middle of constructing pipelines. The failure mode is not fatal though and we only persist after a set number of frames. So I'm not too worried about this.